controllers/factory/vmagent_test.go

```diff
@@ -113,6 +113,28 @@ func TestCreateOrUpdateVMAgent(t *testing.T) {
 				},
 			},
 		},
+		{
+			name: "fail if bearer token secret is missing, without basic auth",
```

Comment:
I can't see any example of using `want` in this method, so since there's `wantErr`, I used negation test here.

I don't think negation test is a good idea as this test is surrounded with non-negation one, but at least this works for now, so might need some help in this part.